onnx/defs/controlflow/defs.cc

```diff
@@ -604,7 +605,15 @@ ONNX_OPERATOR_SET_SCHEMA(
             " if the dimensions or data type of these scan_outputs change across loop"
             " iterations.",
             AttributeProto::GRAPH)
-        .TypeConstraint("V", OpSchema::all_tensor_types(), "All Tensor types")
+        .TypeConstraint(
+            "V",
+            [](){
+              auto t = OpSchema::all_tensor_types();
```

Comment:
See comment above. If we could generalize the Schema implementation so that a type-variable "V" can be unconstrained (that is, without any TypeConstraint specified for it), we could omit this type-constraint. I think it involves some non-trivial work to extend this, but it would be really useful and great.